responsiveslides.js

```diff
@@ -369,6 +371,85 @@
 
       }
 
+      //Touch
+      if (settings.touch) {
+
+        //Borrowed heavily from localpcguy https://gist.github.com/localpcguy/1373518
+        var rsTouch = {
+          touches : {
+            "touchstart": {"x": -1, "y": -1},
+            "touchmove" : {"x": -1, "y": -1},
+            "touchend"  : false,
+            "direction" : "undetermined"
+          },
+          touchHandler: function(e) {
+            var touch;
+
+            //Do not execute if no event
+            if (typeof e !== "undefined") {
+              e.preventDefault();
+
+              //Do not execute if touch values are not returned
+              if (typeof e.originalEvent.touches !== "undefined") {
+                touch = e.originalEvent.touches[0];
+
+                //Filter through for each touch event
+                switch (e.type) {
+                case "touchstart":
+                  //On first touch, pause the scroll so new slide doesn't appear
+                  clearInterval(rotate);
+                case "touchmove":
+                  //Store current touch location
+                  rsTouch.touches[e.type].x = touch.pageX;
+                case "touchend":
+                  //Continue if touch was actually executed
+                  if (rsTouch.touches.touchstart.x > -1 && rsTouch.touches.touchmove.x > -1) {
+                    //Set direction
+                    rsTouch.touches.direction = rsTouch.touches.touchstart.x < rsTouch.touches.touchmove.x ? "right" : "left";
+
+                    //Determine where to slide (duplicated from line 347)
+                    var $visibleClass = $("." + visibleClass);
+                    var idx = $slide.index($visibleClass),
+                      prevIdx = idx - 1,
+                      nextIdx = idx + 1 < length ? index + 1 : 0;
+
+                    //Round up to positive integer, continue if swipe is significant enough (over threshold)
+                    if (Math.abs(rsTouch.touches.touchstart.x - rsTouch.touches.touchmove.x) > settings.touchThreshold) {
+
+                      //The actuall sliding execution
+                      if (rsTouch.touches.direction === "left") {
+                        slideTo(nextIdx);
+                      } else {
+                        slideTo(prevIdx);
```

Comment:
// To update the pager
$('.rslides_tabs li').removeClass("rslides_here");
var slide = 'li.rslides1_s' + (prevIdx + 1);
$(slide).addClass("rslides_here");